src/Apps/Artwork/ArtworkApp.tsx

```diff
@@ -88,12 +90,42 @@ const BelowTheFoldArtworkDetails: React.FC<BelowTheFoldArtworkDetailsProps> = ({
 
 export const ArtworkApp: React.FC<Props> = props => {
   const { artwork, me, referrer, tracking, shouldTrackPageView } = props
-  const { match, silentPush } = useRouter()
+  const { match, silentPush, silentReplace } = useRouter()
+  const { showAuthDialog } = useAuthDialog()
 
   const privateArtworksEnabled = useFeatureFlag(
     "amber_artwork_visibility_unlisted"
   )
 
+  useEffect(() => {
+    const expectingPartnerOffer = !!match?.location?.query?.partner_offer_id
+    const isLoggedIn = !!me
+
+    if (expectingPartnerOffer && typeof window !== "undefined") {
+      if (!isLoggedIn) {
```

Comment:
Short-circuiting `returns` would be a lot more clear than this nested conditional